test/integration/dialects/mssql/connection-manager.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+var chai = require('chai')
+  , expect = chai.expect
+  , Sequelize = require(__dirname + '/../../../../index')
+  , config = require(__dirname + '/../../../config/config').mssql;
+
+config.dialect = 'mssql';
+config.dialectOptions = {};
+config.dialectOptions.domain = 'TEST.COM';
+
+describe('[MSSQL] Connection Manager', function () {
+  var instance;
+  beforeEach(function () {
+    instance = new Sequelize(config.database
+                             , config.username
+                             , config.password
+                             , config);
+  });
+  
+  it('Does not delete `domain` from config.dialectOptions', function (done) {
+    // Commit 91fc76be10 deleted config.dialectOptions.domain after creating a
+    // connection. As a result, every subsequent connection was unable to
+    // authenticate to a domain, since it was passed to `connectionConfig`
+    // as undefined.
+    //
+    // This test will fail authentication to the server, since there is no domain,
+    // but it does confirm that config.dialectOptions.domain persists for
+    // subsequent connections.
+
+    expect(config.dialectOptions.domain).to.equal('TEST.COM');
+    
+    instance.dialect.connectionManager.$connect(config).then(function (res) {
+      expect(config.dialectOptions.domain).to.equal('TEST.COM');
+      done();
+    })
+    .catch(function (err) {
+      // We expect a SequelizeConnectionError
+      if (err.name !== 'SequelizeConnectionError') {
+        throw err;
+      }
+      expect(config.dialectOptions.domain).to.equal('TEST.COM');
+      done();
+    })
+    .catch(function (err) {
+      // err MAY be a Chai AssertionError. Pass it to done.
+      done(err);
+    });
+  });
+});
```

Comment:
Thanks. It looks to me like stubbing out tedious would be, well, very tedious. How about we force the test to point at an unknown db on localhost, and perform the test on ConnectionRefusedError?

```javascript
expect(config.dialectOptions.domain).to.equal('TEST.COM');
       
return instance.dialect.connectionManager.$connect(config)
          .catch(SequelizeConnectionRefusedError, function (err) {
            expect(config.dialectOptions.domain).to.equal('TEST.COM');
          });
```
If that's acceptable, where should I place this test file? If not acceptable, any clues on where to insert a tedious stub?